data_steward/cdr_cleaner/cleaning_rules/deid/generate_site_mappings_and_ext_tables.py

```diff
@@ -0,0 +1,147 @@
+"""
+Rule to create non-deterministic site ids for ehr sites and create the ext_tables
+"""
+# Python Imports
+import logging
+
+# Project imports
+from constants.bq_utils import LOOKUP_TABLES_DATASET_ID, HPO_SITE_ID_MAPPINGS_TABLE_ID
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from tools.generate_ext_tables import get_generate_ext_table_queries, parse_args
+from common import JINJA_ENV
+
+LOGGER = logging.getLogger(__name__)
+
+ISSUE_NUMBERS = ['DC-1351']
+
+SITE_TABLE_ID = '_site_mappings'
+EHR_SITE_PREFIX = 'EHR site '
+RDR = 'rdr'
+PPI_PM = 'PPI/PM'
+
+SITE_MAPPINGS_QUERY = JINJA_ENV.from_string("""
+CREATE TABLE IF NOT EXISTS
+  `{{project_id}}.{{dataset_id}}.{{table_id}}` (hpo_id STRING OPTIONS(description="The hpo_id of the hpo site/RDR."),
+    src_id STRING OPTIONS(description="The masked id of the hpo site/RDR.")) AS (
+    WITH random_ids AS
+(
+  SELECT
+    CONCAT({{site_prefix}}, new_id) AS hpo_id,
+    ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()) AS row_id
+  FROM
+    UNNEST(GENERATE_ARRAY(100, 999)) AS new_id
+)
+SELECT
+  r.hpo_id,
+  LOWER(m.hpo_id) as src_id
+FROM
+(
+  SELECT
+    m.*,
+    ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()) AS row_id
+  FROM `{{project_id}}.{{lookup_tabels_dataset}}.{{hpo_site_id_mappings_table}}` AS m
+  WHERE m.HPO_ID IS NOT NULL AND m.HPO_ID != ''
+) AS m
+JOIN random_ids AS r
+USING (row_id)
+union all
+select {{ppi_pm}} as hpo_id, {{rdr}} as src_id
+)
+""")
+
+
+class GenerateSiteMappingsAndExtTables(BaseCleaningRule):
+    """
+    Generates non-deterministic site_ids for hpo sites.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 mapping_dataset_id):
+        """
+        Initialize the class with proper info.
+
+        Set the issue numbers, description and affected datasets.  As other
+        tickets may affect this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = f'Change PIDs to RIDs in specified tables'
+
+        self.mapping_dataset_id = mapping_dataset_id
```

Comment:
Perhaps set it to a private member, use a getter to get `mapping_dataset_id`, just like how `dataset_id` and other members are handled in the [base class ](https://github.com/all-of-us/curation/blob/3ba9ad65eb4e374bc43797f2bac84ebf17d6ac6a/data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py#L316)